xml_converter/generators/code_generator.py

```diff
@@ -359,13 +388,55 @@ def build_example(self, type: str, applies_to: List[str], xml_field: str, exampl
 
         return example
 
+    ############################################################################
+    # Generate Node Types
+    #
+    # This will output code for a single category of nodes.
+    ############################################################################
+    def generate_node_types(self, metadata: Any, attributenames: Dict[str,str],page: str) -> Tuple[List[str],List[str]]:
+        
+        field_rows: List[str] = []
+        includelist: Set[str] = set()
+        typechange: Dict[str,str] = {
+            "Fixed32": "int",
+            "Int32": "int",
+            "Boolean": "bool",
+            "Float32": "float",
+            "String": "string",
+        }
+        
+        for fieldkey,field in metadata.items():
+            for x in attributenames:
+                if fieldkey in x :
+                    attributename = attributenames[x]
+                   
+            if page in field['applies_to']:
+                if field['type'] in typechange:
+                    cpptype = typechange[field['type']]
+                    includelist.add(cpptype)
+                elif field['type'] == "Custom":
+                    cpptype = field['class']
+                    includelist.add(cpptype.lower())
+                elif field['type'] in ["Enum","MultiflagValue","CompoundValue"]:
+                    cpptype = capitalize(attributename,delimiter="")
+                    includelist.add(attributename)
+                    
+                else :
+                    raise ValueError("Unexpected type {field_type} for attribute {attributename}".format(
+                        field_type=field['type'],
+                        attributename=attributename,
+                    ) )
+               
+                field_rows.append((attributename,cpptype))
```

Comment:
`field_rows` is defined as a `List[str]` but you are appending a `Tuple[str, str]` to it here.

Reply:
`field_rows` is now `attribute_variables` and is redefined as `attribute_variables: List[Tuple[str]] = []`